PhysicsTools/TensorFlow/test/testBase.h

```diff
@@ -20,7 +22,10 @@ class testBase : public CppUnit::TestFixture {
   void tearDown();
   std::string cmsswPath(std::string path);
 
-  virtual void checkAll() = 0;
+  virtual void test(tensorflow::Backend backend) = 0;
+  void checkCPU();
+  void checkGPU();
```

Comment:
Given where we stand towards portability and introduction of support for AMD GPUs, it would be better to move the tests depending on CUDA to separate files that are compiled only if CUDA is present (along
```xml
<iftool name="cuda-gcc-support">
  <bin ...>
</iftool>
```
in `BuildFile.xml`). This approach would generalize naturally (although naively with some copy-paste) to other GPU vendors, and would work on architectures where we don't support CUDA (like GCC 12 currently).

I'd also suggest to use the term `CUDA` explicitly instead of `GPU` on all functions that depend on CUDA specifically.